### DISCUSSION TITLE
The upscale list keeps growing...

### DISCUSSION THREAD

**Discussion by tkiikt** [vladmandic/sdnext#221] [Q&A] [answered]
Hi, I just installed "automatic". For some reason the upscale list keeps growing.I think there were 4 each of "LDSR", "ScuNET GAN", "ScuNET PSNR" and "SwinIR 4x" right after installation.
How can I fix this?

<img width="378" alt="2023-04-19 151102" src="https://user-images.githubusercontent.com/129479703/232984691-b7cc87e6-bace-4e75-9d33-b419ba42ead2.png">

**Comment by tkiikt** [comment, marked as answer]
Sorry. Rebooting fixed it.

**Comment by tkiikt** [comment]
Now I know why. Change the theme and press the apply setting button -> restart ui button.
Then LDSR, ScuNET GAN, ScuNET PSNR and SwinIR 4x will increase.

**Comment by vladmandic** [comment]
using restart server from ui causes reload of all scripts/extensions - and quite a few don't check if they're already registered, so they register themselves and create duplicates. same root cause leads to a lot of different problems with restart server from ui (random `keyError` messages in console - that's because key ids are no longer the same because of duplicates). only way to fix that is to go to each script/extension and add "if registered" clause.

**Comment by vladmandic** [comment]
see #224 for details.

**Comment by tkiikt** [comment]
ok, i see. thank you!
